### PR TITLE
SNOW-1763420 Allow creating version with only MANAGE VERSIONS privilege

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
If a user only has the following permissions:
- `MONITOR` on the package
- `USAGE` on the stage’s schema
- `READ` on the stage
- `MANAGE VERSIONS` on the package
They should still be able to create a new version. 

To remove the requirement to have privilege to create a stage, we now check if the stage exists instead of doing `create stage if not exists`. To remove the requirement to have privilege to check for release directives, we now show a different warning if we couldn't check whether a version is already included in a release directive.

### CODE REVIEW

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py [hunk 1]
```diff
@@ -206,3 +248,121 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
...
+            except ProgrammingError as e:
+                if e.errno == INSUFFICIENT_PRIVILEGES:
+                    raise InsufficientPrivilegesError(
+                        f"Insufficient privileges to create stage {name}",
+                        role=role,
+                        database=database,
+                        schema=schema,
+                    ) from e
+                raise
+
```
Comment (sfc-gh-pjafari) [inline src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py:347]: make sure to use `handle_unclassified_error` for all the other exceptions that are caught from the facade.

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py [hunk 2]
```diff
@@ -181,6 +186,129 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
             return None
         return table
 
+    def create_schema(
```
Comment (sfc-gh-melnacouzi) [inline src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py:189]: we are missing tests for these

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -33,6 +33,7 @@
   Removed incorrect copying of connections from `connections.toml` to `config.toml`.
 * Fixes `snow connection generate-jwt` to work with keys with no passphrase.
 * `snow connection add` supports `--no-interactive` flag to skip interactive prompts.
+* The privilege to create a schema or stage is no longer required to run `snow app version create`.
```
Comment (sfc-gh-bdufour) [inline RELEASE-NOTES.md:36]: do we need to specify that this is only if the schema/stage already exist?

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py [hunk 4]
```diff
@@ -181,6 +186,129 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
...
+        self, name: str, role: str | None = None, database: str | None = None
+    ):
+        """
+        Creates a schema.
+        @param name: Name of the schema to create. Can be a fully qualified name or just the schema name, in which case the current database or the database passed in will be used.
+        @param [Optional] role: Role to switch to while running this script. Current role will be used if no role is passed in.
+        @param [Optional] database: Database to use while running this script.
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py:196]: it's used as the default, but this seems to indicate that it's an override. We should align the description and the implementation (either rename this to default_database or use it as an override).

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py [hunk 5]
```diff
@@ -181,6 +186,129 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
...
+            try:
+                self._sql_executor.execute_query(
+                    f"create schema if not exists {identifier}"
+                )
+            except ProgrammingError as err:
+                if err.errno == INSUFFICIENT_PRIVILEGES:
+                    raise InsufficientPrivilegesError(
+                        f"Insufficient privileges to create schema {name}",
+                        role=role,
+                        database=database,
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py:213]: that's not always the DB we're using

File: tests/nativeapp/test_sf_sql_facade.py [hunk 6]
```diff
@@ -53,6 +56,90 @@ def reset_sql_facade():
...
+    mock_cms: list[tuple[mock.Mock, Call]],
+    inner_mocks: list[tuple[mock.Mock, Call]],
+):
+    """Assert that certain calls are made within a series of context managers.
+
+    Use it like so:
+
+    expected_inner_calls = [mock.call("select 1"), mock.call("select 2")]
+    mock_cms = [(mock_use_role, mock.call("test_role")), (mock_use_warehouse, mock.call("test_wh"))]
+    with assert_within_use_calls(expected_inner_calls, mock_execute_query, mock_cms):
```
Comment (sfc-gh-bdufour) [inline tests/nativeapp/test_sf_sql_facade.py:100]:
```suggestion
    with assert_in_context(mock_cms, expected_inner_calls):
```

**Review by sfc-gh-bdufour — DISMISSED** [review]

File: RELEASE-NOTES.md (hunk 3 above)
Comment (sfc-gh-fcampbell) [reply RELEASE-NOTES.md]: updated

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py (hunk 4 above)
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py]: updated

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py (hunk 5 above)
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py:378]: fixed

File: tests/nativeapp/test_sf_sql_facade.py (hunk 6 above)
Comment (sfc-gh-fcampbell) [reply tests/nativeapp/test_sf_sql_facade.py]: fixed

File: src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py (hunk 1 above)
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py]: done

**Review by sfc-gh-bdufour — APPROVED** [review]

**Pull request merged**